NOTES.md

```diff
@@ -0,0 +1,270 @@
+# Librix server API Guide _18/05/2021_
+
+Running locally is set to port 5000.
+
+## Table of Contents
+
+- [Layout](#layout)
+- [Errors](#errors)
+- [Users](#users)
+- [Books](#books)
+- [Matches](#matches)
+
+## Layout
+
+The layout of the sections below will be as follows:
+
+- After the main heading will be the routes, eg. _/_ or _/:id_
+- A brief description about what the call does, eg. _Add User_
+- In those routes will be the calls avaiable, eg. _get_ or _post_
+- In those calls will be one or many of the following: _Frontend Request_, _Backend Response_ and _Notes_
+- **Notes** can about the Frontend or the Backend depending on what the come after, eg.
+
+```
+_Frontend Request_
+_Notes_ (specificlly about the Frontend)
+_Backend Response_
+_Notes_ (specifically about the Backend)
+```
+
+## Errors
+
+_Notes_ => If there is an error in any of the functions we return =>
+
+```
+{
+  "error": {
+    "message": "The error message"
+  }
+}
+```
+
+## Users
+
+### /users/
+
+- Add User
+  **post**
+  _Frontend Request_ =>
+
+  ```
+  {
+    "username": "username",
+    "email": "email@email.com",
+    "password": "password",
+    "address": {
+      "city": "city"
+    }
+  }
+  ```
+
+  _Notes_ => That was the minimal request required, other fields that can be added include =>
+
+  ```
+  {
+    "firstName": "firstName",
+    "lastName": "lastName",
+    "avatar": "avatar",
+    "address": {
+      "country": "country"
```

Comment:
avatar is fallback if not loaded up right?

Reply:
yes, can the user add their own avatar or only on update?

Reply:
done